slurmutils/models/option.py

```diff
@@ -48,6 +49,30 @@ def keys(cls) -> Iterable[str]:
             yield field.name
 
 
+@dataclass(frozen=True)
+class AcctGatherConfigOptionSet(_OptionSet):
+    """`acct_gather.conf` configuration options."""
+
+    EnergyIPMIFrequency: Callback = Callback()
+    EnergyIPMICalcAdjustment: Callback = Callback()
+    EnergyIPMIPowerSensors: Callback = Callback()
```

Comment:
Looks like [`EnergyIMPIPowerSensors`](https://slurm.schedmd.com/acct_gather.conf.html#OPT_EnergyIPMIPowerSensors) requires some special parsing as it's like a Slurm dictionary but uses `;` as the separator rather than `:`. Easiest thing we can do here is just create a custom `EnergyIMPIPowerSensors` callback in _callback.py_:

```python3
def from_energy_impi_power_sensors(value: str) -> Dict[str, List[str]]:
    result = {}
    for opt in value.split(";"):
        k, v = opt.split("=", maxsplit=1)
        result[k] = v.split(",")

    return result

def to_energy_impi_power_sensors(value: Dict[str, List[str]]) -> str:
    result = []
    for k, v in value.items():
        result.append(f"{k}={','.join(v)}")

    return ";".join(result)

EnergyIMPIPowerSensorsCallback(from_energy_impi_power_senors, to_energy_impi_power_sensors)
```

With this callback we can easily manipulate the value of `EnergyIMPIPowerSensors` from Python, but still have the value in the format that Slurm prefers. Note that you'll need to up the corresponding unit test once we implement this callback :sweat_smile: 

> One thing I might do is take a harder look at the SlurmDictCallback to see if we can generalize it such that we don't need a custom parsing & marshaling callback for `EnergyIMPIPowerSensors`, but that's something to hack and wack on before releasing slurmutils 0.9.0.